`Masked Input` Copy button guidelines (#1581)
hashicorp/design-system: website/docs/components/form/masked-input/partials/guidelines/guidelines.md

  <F.Label>Label</F.Label>
</Hds::Form::MaskedInput::Field>

## Copyable masked input 

> The `MaskedInput` component lets you show a Copy Button by setting the `hasCopyButton` property to `true` Users can interact with this button to copy the `MaskedInput's` content to their clipboard. Refer to the [copy button](www.helios.com/copy/copy-button) usage guidelines to learn more about copying content. 

Reviewer: Couple of broad notes here:
- Generally when referring to the component, we've moved away from wrapping the name in backticks and instead reserve that styling for properties, arguments, etc.
- When linking to another document/page you don't need the entire URL, all you need is the local path, which in this case is `/components/copy/copy-button`.

```suggestion
The Masked Input component lets you show a Copy Button by setting the `hasCopyButton` property to `true` Users can interact with this button to copy the Masked Input's content to their clipboard. Refer to the [Copy Button](/components/copy/copy-button) usage guidelines to learn more about copying content. 
```

Reviewer: Additionally, I might recommend re-wording this a little bit. Maybe consider:

Setting the `hasCopyButton` property to `true` will display a [Copy Button](/components/copy/copy-button) adjacent to the Masked Input, allowing users to copy the value in the Masked Input to their clipboard. Refer to the [Copy Button](/components/copy/copy-button) usage guidelines for more details around copying content.

Author: Ah, okay! Thank you for clarifying.

> The `MaskedInput` component lets you show a Copy Button by setting the `hasCopyButton` property to `true` Users can interact with this button to copy the `MaskedInput's` content to their clipboard. Refer to the [copy button](www.helios.com/copy/copy-button) usage guidelines to learn more about copying content. 

Author: ```suggestion
Setting the hasCopyButton property to true will display a [Copy Button](components/copy/copy-button) adjacent to the Masked Input, allowing users to copy the value in the Masked Input to their clipboard. Refer to the [Copy Button](components/copy/copy-button) usage guidelines for more details around copying content. 
```

> The new boolean property allows for toggling the display of a Copy Button on the right side of the MaskedInput. When set to `true,` the Copy Button becomes visible, helping users to copy the masked content to their clipboard. Refer to the [copybutton](www.helios.com/copy/copy-button) usage guidelines to learn more about copying content. 

Author: ```suggestion
The `MaskedInput` component lets you show a Copy Button by setting the `hasCopyButton` property to `true` Users can interact with this button to copy the `MaskedInput's` content to their clipboard. Refer to the [copy button](www.helios.com/copy/copy-button) usage guidelines to learn more about copying content. 
```

> ![Masked input with a copy button](/assets/components/form/masked-input/masked-input-has-copy-button.png)

Reviewer: Just for consistency in capitalization, even though this isn't visible text.

```suggestion
![Masked Input with a Copy Button](/assets/components/form/masked-input/masked-input-has-copy-button.png)
```

> Setting the hasCopyButton property to true will display a [Copy Button](components/copy/copy-button) adjacent to the Masked Input, allowing users to copy the value in the Masked Input to their clipboard. Refer to the [Copy Button](components/copy/copy-button) usage guidelines for more details around copying content. 
> 
> ![Masked Input with a Copy Button](/assets/components/form/masked-input/masked-input-has-copy-button.png)

Author: ```suggestion
Setting the `hasCopyButton` property to `true` will display a [Copy Button](components/copy/copy-button) adjacent to the Masked Input, allowing users to copy the value in the Masked Input to their clipboard. Refer to the [Copy Button](components/copy/copy-button) usage guidelines for more details around copying content. 

![Masked Input with a Copy Button](/assets/components/form/masked-input/masked-input-has-copy-button.png)
```

> ![Masked Input with a Copy Button](/assets/components/form/masked-input/masked-input-has-copy-button.png)

Reviewer 2: Using the actual component to show the functionality may be more impactful because they can see it in action. I realize it's not available in code yet, but the docs can't be merged until they're available anyway.

Once ready, you should be able to use the following code: 

```suggestion
<Doc::Layout @spacing="16px">
  <Hds::Form::MaskedInput::Field @value="0362df4996ca864b4df9e42cba0d6d" @width="300px" @hasCopyButton={{true}} as |F|>
    <F.Label>Label</F.Label>
  </Hds::Form::MaskedInput::Field>
  <Hds::Form::MaskedInput::Field @value="0362df4996ca864b4df9e42cba0d6d" @isMultiline={{true}} @width="300px" @hasCopyButton={{true}} as |F|>
    <F.Label>Label</F.Label>
  </Hds::Form::MaskedInput::Field>
</Doc::Layout>
```

@alex-ju can you double-check that this code would be accurate?

Reviewer 3: yes, that's how it should look like. Haven't got the chance to work on the code yet, but I'll try and make it a priority to avoid this PR stalling for too long.

Reviewer 2: Thanks @alex-ju!

## Error validation

> For error validation recommendations, refer to the [Form patterns](/patterns/form-patterns) documentation

Reviewer: Not sure why the period was removed here, maybe in error?

Author: Hmm, yeah not sure 🧐

> For error validation recommendations, refer to the [Form patterns](/patterns/form-patterns) documentation

Author: ```suggestion
For error validation recommendations, refer to the [Form patterns](/patterns/form-patterns) documentation.
```

  <F.Label>Label</F.Label>
</Hds::Form::MaskedInput::Field>

> ## Copyable masked input 

Reviewer: Not overly opinionated here, but "Copyable" isn't really sitting right with me. Maybe consider changing this headline to something like "Copying the Masked Input" (also note the capitalization when the component name is being referenced). Some other options while I'm at it:

- Copying the Masked Input value
- Masked Input copying

Reviewer 2: +1, I like "Copying the Masked Input value" personally. It's the most clear.

> ## Copyable masked input 

Author: ```suggestion
## Copying the Masked Input value
```
